docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
+
+The other packages are described below: 
+1. `UI` : The user interface of the app, reads user input and is visible to the user.
+1. `Parser Manager` : The module that reads user inputs, and creates a suitable parser based on the command to make 
+ sense of user input. Respective parser then tells the Command module what to execute.
+1. `ProjectManager` : Manages and stores all the projects added by the user, keeps track of selected project.    
+1. `SprintManager` : Stores all sprints associated with a project.
+1. `TaskManager` : Stores all tasks in backlog associated with a project.
+1. `ProjectMembers` : Stores all team-members associated to a project. 
+1. `Parser` : Creates a suitable parser, based on the command to make sense of the user input. Respective parser then
+ make use the information and call respective commands. 
+ 
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
+
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
+### Logic Component
+![Figure X: Simplified class diagram for Logic Component](./image/developerguide/parserandcommand.png)
+
+When a user types a command, `SCRUMptious`calls the `ParserManager`. The `ParserManager` then parses commands from the user. Subsequently, 
+the `ParserManager` passes the commands on to the respective exceptions parsers which inherit from the `ExceptionsParser` interface. 
+The exceptions parsers consist of:  
+- `ProjectParser`
+- `MemberParser`
+- `TaskParser`
+- `SprintParser`
+- `HelpParser`
+
+The `ProjectParser` validates the parameters of the command. If the command is valid, it returns the respective `ProjectCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `ProjectParser` returns an appropriate warning message to the user.  
+
+The `MemberParser` validates the parameters of the command. If the command is valid, it returns the respective `MemberCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `TaskParser` validates the parameters of the command. If the command is valid, it returns the respective `TaskCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `SprintParser` validates the parameters of the command. If the command is valid, it returns the respective `SprintCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `SprintParser` returns an appropriate warning message to the user.  
+
+The `HelpParser` validates the parameters of the command. If the command is valid, it returns the respective `HelpParserCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `HelpParser` returns an appropriate warning message to the user.
```

Comment:
I like the way this is styled. However I think it would be better if the different "xxxParser" exceptions parsers were split into different subheadings.